nemo/utils/model_utils.py

```diff
@@ -145,6 +146,31 @@ def parse_dataset_as_name(name: str) -> str:
     return name
 
 
+def unique_names_check(name_list: Optional[List[str]]):
+    """
+    Performs a uniqueness check on the name list resolved, so that it can warn users
+    about non-unique keys.
+
+    Args:
+        name_list: List of strings resolved for data loaders.
+    """
+    if name_list is None:
+        return
+
+    # Name uniqueness checks
+    names = {}
```

Comment:
Why not use a set object instead of a dictionary?

Reply:
Good point, forgot that sets have O(1) lookup too. Done.